### PR TITLE
Refactor/rename messaging contribution push param

### PR DESCRIPTION
**Pull request by harbin1053020115**
#### What it does

Rename message-contirbution.ts(packages/core/src/node/messaging/messaging-contribution.ts) ConnectionHandlers.push param, which is changing channel to socket.

#### How to test

No need to test.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by vince-fugnitto — COMMENTED** [review]
Thank you for your contribution :+1: 

In order to accept your changes please be sure to sure to sign the [eclipse contributor agreement (eca)](https://github.com/eclipse-theia/theia/blob/master/CONTRIBUTING.md#eclipse-contributor-agreement) with the same email as your authorship.

I'm not sure I fully understand the reasons for the change, it is only updating the variable name in the callback, uses of `MessagingContribution.push` still use `channel` (verified with `find all references`).

**Comment by harbin1053020115** [comment]
> Thank you for your contribution 👍
> 
> In order to accept your changes please be sure to sure to sign the [eclipse contributor agreement (eca)](https://github.com/eclipse-theia/theia/blob/master/CONTRIBUTING.md#eclipse-contributor-agreement) with the same email as your authorship.
> 
> I'm not sure I fully understand the reasons for the change, it is only updating the variable name in the callback, uses of `MessagingContribution.push` still use `channel` (verified with `find all references`).

The original `channel` param name is misunderstanding with channels concept in MessagingContribution.handleChannels method.

**Comment by vince-fugnitto** [comment]
> The original `channel` param name is misunderstanding with the channel concept in MessagingContribution.handleChannels method, but actually it means `socket`.

I believe `socket` might also be confusing since it accepts any generic `connection` (likely the best name).
Currently we have `WebSocketChannel` and `Socket` that make use of `ConnectionHandlers`:

https://github.com/eclipse-theia/theia/blob/3893fdec2fc07766da986e12b62457a897b3b64e/packages/core/src/node/messaging/messaging-contribution.ts#L55

https://github.com/eclipse-theia/theia/blob/3893fdec2fc07766da986e12b62457a897b3b64e/packages/core/src/node/messaging/messaging-contribution.ts#L56

https://github.com/eclipse-theia/theia/blob/3893fdec2fc07766da986e12b62457a897b3b64e/packages/core/src/electron-main/messaging/electron-messaging-contribution.ts#L46

**Comment by harbin1053020115** [comment]
> 


> > The original `channel` param name is misunderstanding with the channel concept in MessagingContribution.handleChannels method, but actually it means `socket`.
> 
> I believe `socket` might also be confusing since it accepts any generic `connection` (likely the best name). Currently we have `WebSocketChannel` and `Socket` that make use of `ConnectionHandlers`:
> 
> https://github.com/eclipse-theia/theia/blob/3893fdec2fc07766da986e12b62457a897b3b64e/packages/core/src/node/messaging/messaging-contribution.ts#L55
> 
> https://github.com/eclipse-theia/theia/blob/3893fdec2fc07766da986e12b62457a897b3b64e/packages/core/src/node/messaging/messaging-contribution.ts#L56
> 
> https://github.com/eclipse-theia/theia/blob/3893fdec2fc07766da986e12b62457a897b3b64e/packages/core/src/electron-main/messaging/electron-messaging-contribution.ts#L46



> > The original `channel` param name is misunderstanding with the channel concept in MessagingContribution.handleChannels method, but actually it means `socket`.
> 
> I believe `socket` might also be confusing since it accepts any generic `connection` (likely the best name). Currently we have `WebSocketChannel` and `Socket` that make use of `ConnectionHandlers`:
> 
> https://github.com/eclipse-theia/theia/blob/3893fdec2fc07766da986e12b62457a897b3b64e/packages/core/src/node/messaging/messaging-contribution.ts#L55
> 
> https://github.com/eclipse-theia/theia/blob/3893fdec2fc07766da986e12b62457a897b3b64e/packages/core/src/node/messaging/messaging-contribution.ts#L56
> 
> https://github.com/eclipse-theia/theia/blob/3893fdec2fc07766da986e12b62457a897b3b64e/packages/core/src/electron-main/messaging/electron-messaging-contribution.ts#L46

so, if this is meaningful:
```typescript
push(spec: string, callback: (params: MessagingService.PathParams, connection: T) => void): void {
    const route = new Route(spec);
    // here is the change, or you can look at the lastest changes.
    this.handlers.push((path, channel: T) => {
        const params = route.match(path);
        if (!params) {
            return false;
        }
        callback(params, channel);
        return route.reverse(params);
    });
}
```

File: packages/core/src/node/messaging/messaging-contribution.ts [hunk 1]
```diff
@@ -211,7 +211,7 @@ export namespace MessagingContribution {
 
         push(spec: string, callback: (params: MessagingService.PathParams, connection: T) => void): void {
             const route = new Route(spec);
-            this.handlers.push((path, channel) => {
+            this.handlers.push((path, channel: T) => {
```
[the remark this answers is not in the document]
Comment (harbin1053020115) [reply packages/core/src/node/messaging/messaging-contribution.ts:214]: > 

I see, please ignore this pr

[the remark this answers is not in the document]
Comment (vince-fugnitto) [reply packages/core/src/node/messaging/messaging-contribution.ts:214]: No problem! If there are improvements or features you'd like to work on please don't hesitate to contribute and we'd be happy to review them :)

**Pull request closed without merging**